src/openms/include/OpenMS/CHEMISTRY/MASSDECOMPOSITION/IMS/RealMassDecomposer.h

```diff
@@ -126,7 +126,7 @@ namespace OpenMS
         Decomposer to be used for exact decomposing using
         integer arithmetic.
       */
-      std::auto_ptr<integer_decomposer_type> decomposer_;
+      std::shared_ptr<integer_decomposer_type> decomposer_;
```

Comment:
not sure about the copy constructor here, it may fail when building a copy constructor as `unique_ptr` cannot be copied.